### ISSUE TITLE
ECONNREFUSED error causes task failure when AS3 fetches external resources

### ISSUE THREAD

**Issue by andreykashcheev** [F5Networks/f5-appsvcs-extension#455]
### Environment
 * Application Services Version: 3.26.0
 * BIG-IP Version: 16.0.1

### Summary
Please also include information about the reproducibility and the severity/impact of the issue.

AS3 allows to reference external resources (i.e. WAF Policy) within declaration. In a case when attempt to fetch external resource ends up with ECONNREFUSED error, AS3 fails the task with status code 422 (Unprocessable Entity) however re-attempt to fetch the same external resource manually works as expected. It indicates that enabling retry logic on none-transient networking errors would resolve/mitigate the task failures. 

This is intermittent issue but it fails BIGIP on-boarding. 

### Steps To Reproduce
1. Submit the following declaration:
```json
{
  "schemaVersion": "3.0.0",
  "class": "ADC",
  "remark": "Autoscale",
  "label": "Autoscale",
  "Tenant_1": {
    "class": "Tenant",
    "Shared": {
      "class": "Application",
      "template": "shared",
      "telemetry_local_rule": {
        "remark": "Only required when TS is a local listener",
        "class": "iRule",
        "iRule": "when CLIENT_ACCEPTED {\n  node 127.0.0.1 6514\n}"
      },
      "telemetry_local": {
        "remark": "Only required when TS is a local listener",
        "class": "Service_TCP",
        "virtualAddresses": [
          "255.255.255.254"
        ],
        "virtualPort": 6514,
        "iRules": [
          "telemetry_local_rule"
        ]
      },
      "telemetry": {
        "class": "Pool",
        "members": [
          {
            "enable": true,
            "serverAddresses": [
              "255.255.255.254"
            ],
            "servicePort": 6514
          }
        ],
        "monitors": [
          {
            "bigip": "/Common/tcp"
          }
        ]
      },
      "telemetry_hsl": {
        "class": "Log_Destination",
        "type": "remote-high-speed-log",
        "protocol": "tcp",
        "pool": {
          "use": "telemetry"
        }
      },
      "telemetry_formatted": {
        "class": "Log_Destination",
        "type": "splunk",
        "forwardTo": {
          "use": "telemetry_hsl"
        }
      },
      "telemetry_publisher": {
        "class": "Log_Publisher",
        "destinations": [
          {
            "use": "telemetry_formatted"
          }
        ]
      },
      "telemetry_asm_security_log_profile": {
        "class": "Security_Log_Profile",
        "application": {
          "localStorage": false,
          "remoteStorage": "splunk",
          "servers": [
            {
              "address": "255.255.255.254",
              "port": "6514"
            }
          ],
          "storageFilter": {
            "requestType": "illegal"
          }
        }
      },
      "shared_pool": {
        "class": "Pool",
        "remark": "Service 1 shared pool",
        "members": [
          {
            "addressDiscovery": "azure",
            "addressRealm": "private",
            "resourceGroup": "ak-runtime-init-hangup",
            "resourceId": "aktest-app-vmss",
            "resourceType": "scaleSet",
            "servicePort": 80,
            "subscriptionId": "<SUBSCRIPTON_ID>",
            "updateInterval": 60,
            "useManagedIdentity": true
          }
        ],
        "monitors": [
          "http"
        ]
      }
    },
    "HTTP_Service": {
      "template": "http",
      "class": "Application",
      "serviceMain": {
        "class": "Service_HTTP",
        "virtualAddresses": [
          "0.0.0.0"
        ],
        "snat": "auto",
        "policyWAF": {
          "use": "WAFPolicy"
        },
        "securityLogProfiles": [
          {
            "use": "/Tenant_1/Shared/telemetry_asm_security_log_profile"
          }
        ],
        "pool": "/Tenant_1/Shared/shared_pool"
      },
      "WAFPolicy": {
        "ignoreChanges": false,
        "url": "https://raw.githubusercontent.com/f5devcentral/f5-asm-policy-templates/master/generic_ready_template/Rapid_Depolyment_Policy_13_1.xml",
        "enforcementMode": "blocking",
        "class": "WAF_Policy"
      }
    },
    "HTTPS_Service": {
      "template": "https",
      "class": "Application",
      "serviceMain": {
        "virtualAddresses": [
          "0.0.0.0"
        ],
        "snat": "auto",
        "policyWAF": {
          "use": "WAFPolicy"
        },
        "securityLogProfiles": [
          {
            "use": "/Tenant_1/Shared/telemetry_asm_security_log_profile"
          }
        ],
        "pool": "/Tenant_1/Shared/shared_pool",
        "serverTLS": {
          "bigip": "/Common/clientssl"
        },
        "redirect80": false,
        "class": "Service_HTTPS"
      },
      "WAFPolicy": {
        "url": "https://raw.githubusercontent.com/f5devcentral/f5-asm-policy-templates/master/generic_ready_template/Rapid_Depolyment_Policy_13_1.xml",
        "ignoreChanges": false,
        "enforcementMode": "blocking",
        "class": "WAF_Policy"
      }
    }
  }
}
```

2. Observe the following error response:
```json
{
  "results": [
    {
      "code": "ECONNREFUSED",
      "message": "declaration failed",
      "response": "GET https://raw.githubusercontent.com/f5devcentral/f5-asm-policy-templates/master/generic_ready_template/Rapid_Depolyment_Policy_13_1.xml get asm policy WAFPolicy from url failed (connect ECONNREFUSED 185.199.108.133:443)",
      "host": "localhost",
      "tenant": "Tenant_1",
      "runTime": 12340
    }
  ],
  "declaration": {
    "schemaVersion": "3.0.0",
    "class": "ADC",
    "remark": "Autoscale",
    "label": "Autoscale",
    "id": "autogen_1e194c37-fc56-4e1e-b01b-abb042f3b965",
    "updateMode": "selective",
    "controls": {
      "archiveTimestamp": "2021-05-05T21:13:47.822Z"
    }
  },
  "code": 422
}
```

### Expected Behavior

When ECONNREFUSED received, AS3 retries request to fetch external resource

### Actual Behavior
A clear and concise description of what actually happens.
Please include any applicable error output.

When ECONNREFUSED received, AS3 fails task as 422 (Unprocessable Entity)

**Comment by dstokesf5** [comment]
I am switching this over to a feature request since the current behavior, while basic, is functioning as intended. I agree that adding some retry capabilities could be helpful for a smoother user experience.

**Comment by dstokesf5** [comment]
This issue has been resolved with the 3.30.0 release of AS3.
